File: ruby/test/task_master/task_repository/csv_adapter_test.rb
require_relative "../../test_helper"

class TaskMaster::TaskRepository::CsvAdapterTest < Minitest::Test
  def setup
    @fixture_file = File.join(__dir__, "../..", "fixtures/test_data.csv")
    @class_name = TaskMaster::TaskRepository::CsvAdapter
  end

  def test_should_list_all_task_in_supplied_csv
    expected = 6
    assert_equal(expected, @class_name.new(@fixture_file).list.count)
  end
end

File: javascript/test/TaskRepository/CsvAdapter.test.js
const path = require('path');
const CsvAdapter = require(path.join(__dirname, '../../lib/taskRepository/CsvAdapter'));

test("should list all task in supplied csv", () => {
	const fixtureFile = path.join(__dirname, "../fixtures/test_data.csv");
	const instance = new CsvAdapter(fixtureFile);
	expect(instance.list().length).toBe(6);
});
